crates/transport-ipc/src/lib.rs

```diff
@@ -113,19 +112,17 @@ impl IpcBackend {
 pub struct ReadJsonStream<T> {
     /// The underlying reader.
     #[pin]
-    reader: BufReader<T>,
-    /// A buffer of bytes read from the reader.
+    reader: tokio_util::compat::Compat<T>,
```

Comment:
TIL, although, what a name lol

Reply:
I previously wasted a few hours on this myself -.-

same with finding a properly maintained `tokio::codec` `Json` implementation.
using SerializeStream is an adequate solution I think